### PR TITLE
JCL/refactor/feature/game stats methods

### PR DESCRIPTION
**Pull request by JCL461437**
Added the final methods for the fubol Game Statistics module along with corresponding tests and fixing merge conflict between Lance's methods and the ones added in this branch in the module.

### CODE REVIEW

File: lib/game_stats.rb [hunk 1]
```diff
@@ -1,7 +1,4 @@
 
-
```
Comment (cheeleertr) [inline lib/game_stats.rb:2]: Cleaning up unused space is good

**Review by LancePants97 — APPROVED** [review]

File: data/games_fixture.csv [hunk 2]
```diff
@@ -9,6 +9,11 @@ game_id,season,type,date_time,away_team_id,home_team_id,away_goals,home_goals,ve
...
+2016030171,20162017,Postseason,4/14/17,20,24,2,3,Rio Tinto Stadium,/api/v1/venues/null
+2016030172,20162017,Postseason,4/16/17,20,24,2,3,Rio Tinto Stadium,/api/v1/venues/null
+2016030173,20162017,Postseason,4/18/17,24,20,3,2,BMO Field,/api/v1/venues/null
 2016030174,20162017,Postseason,4/20/17,24,20,3,1,BMO Field,/api/v1/venues/null
-2014030411,20142015,Postseason,6/4/15,16,14,2,1,Audi Field,/api/v1/venues/null
\ No newline at end of file
+2014030411,20142015,Postseason,6/4/15,16,14,2,1,Audi Field,/api/v1/venues/null
+2014030412,20142015,Postseason,6/6/15,16,14,3,2,Audi Field,/api/v1/venues/null
+2014030413,20142015,Postseason,6/9/15,14,16,3,2,Gillette Stadium,/api/v1/venues/null
+2012030121,20122013,Postseason,5/2/13,9,8,2,2,Red Bull Arena,/api/v1/venues/null
```
Comment (jdmchugh111) [inline data/games_fixture.csv:19]: think this is a great place to land on for the games fixture

File: spec/game_stats_spec.rb [hunk 3]
```diff
@@ -16,6 +9,25 @@
...
+            expect(@stat_tracker.average_goals_per_game).to eq (4.06) #rounded from 4.0588...
+        end
+    end
+
+    describe "#average_goals_by_season" do
+        it "can determine average goals of all games in a given season" do
+            expect(@stat_tracker.average_goals_by_season).to eq ({"20122013" => 3.7, "20162017" => 4.75, "20142015" => 4.33 })
+        end
+    end
+
```
Comment (jdmchugh111) [inline spec/game_stats_spec.rb:30]: awesome work! pumped to get this locked in

**Review by jdmchugh111 — APPROVED** [review]

File: data/games_fixture.csv (hunk 2 above)
Comment (JCL461437) [reply data/games_fixture.csv:19]: Thanks for the guidance with that

**Pull request merged**